lib/OpenQA/Cache.pm

```diff
@@ -0,0 +1,182 @@
+package OpenQA::Cache;
+use strict;
+use warnings;
+
+use File::Basename;
+use Fcntl qw(:flock);
+use Mojo::UserAgent;
+use OpenQA::Utils qw(log_error log_info log_debug);
+use OpenQA::Worker::Common;
+use List::MoreUtils;
+use Data::Dumper;
+use JSON;
+
+require Exporter;
+our (@ISA, @EXPORT);
+@ISA    = qw(Exporter);
+@EXPORT = qw(get_asset);
+
+my $cache;
+my $host;
+my $limit   = 50;
+my $remove  = 1;
+my $db_file = "../cache.db";
+
+sub get {
+    my ($asset) = @_;
+    my $index = List::MoreUtils::first_index { $_ eq $asset } @{$cache->{$host}};
+    splice @{$cache->{$host}}, $index, 1 if @{$cache->{$host}} > 0;
+    unshift @{$cache->{$host}}, $asset;
+    expire_asset();
+}
+
+sub set {
+    my ($asset) = @_;
+    unshift @{$cache->{$host}}, $asset;
+    expire_asset();
+}
+
+sub init {
+    my $class;
+    $host = shift;
+    @{$cache->{$host}} = read_db();
+    log_debug(__PACKAGE__ . ": Initialized with $host");
+}
+
+sub update_setup_status {
+    my $id = @_;
+    my $status = {setup => 1};
+    api_call(
+        'post',
+        'jobs/' . $job->{id} . '/status',
+        json     => {status => $status},
+        callback => "no",
+    );
+    log_debug("Update status so job is not considered dead.");
+}
+
+sub download_asset {
+    my ($id, $type, $asset) = @_;
+
+    open(my $log, '>>', "autoinst-log.txt") or die("Cannot open autoinst-log.txt");
+    local $| = 1;
+    print $log "CACHE: Locking $asset";
+
+    print $log "Attemping to download: $host $asset, $type, $id";
```

Comment:
We do, but here he is logging into autoinst log instead of worker journal. Which I don't know yet if is correct. Note that when job is started autoinst-log is truncated.